### PR TITLE
Finalize VAPWAT keyword

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by totto82** [comment]
@goncalvesmachadoc  Can you rebase and resolve the conflicts?

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this opm-material=429 opm-models=678 opm-simulators=3715 please

**Comment by goncalvesmachadoc** [comment]
jenkins build this opm-material=429 opm-models=678 opm-simulators=3715 please

File: tests/parser/SimulationConfigTest.cpp [hunk 1]
```diff
@@ -106,6 +106,15 @@ const std::string& inputStr_vap_dis = "RUNSPEC\n"
                                       "REGIONS\n"
                                       "\n";
 
+const std::string& inputStr_watvap = "RUNSPEC\n"
```
Comment (joakim-hove) [inline tests/parser/SimulationConfigTest.cpp:109]: Just a tip for the next time - for all new test strings of this type we try to use multiline string literals:
```C++
const std::string inputStr = R"(
RUNSPEC
VAPWAT
DIMENS
 10 3 4 /
GRID
REGIONS
)";
```
then you do not have to fiddle so much with `"` and `\n`.

File: tests/parser/SimulationConfigTest.cpp [hunk 2]
```diff
@@ -232,6 +241,22 @@ BOOST_AUTO_TEST_CASE(SimulationConfig_VAPOIL_DISGAS) {
...
+    auto deck = createDeck(inputStr);
+    TableManager tm(deck);
+    EclipseGrid eg(10, 3, 4);
+    FieldPropsManager fp(deck, Phases{true, true, true}, eg, tm);
+    SimulationConfig simulationConfig(false, deck, fp);
+    BOOST_CHECK_EQUAL( false , simulationConfig.hasVAPWAT());
+
+    auto deck_vd = createDeck(inputStr_watvap);
+    TableManager tm_vd(deck_vd);
+    EclipseGrid eg_vd(10, 3, 4);
```
Comment (joakim-hove) [inline tests/parser/SimulationConfigTest.cpp:254]: Can just reuse the same `EclipseGrid` instance?

Comment (goncalvesmachadoc) [reply tests/parser/SimulationConfigTest.cpp]: Yes, thank you. Just deleted eg_vd.

File: tests/parser/SimulationConfigTest.cpp (hunk 1 above)
Comment (goncalvesmachadoc) [reply tests/parser/SimulationConfigTest.cpp]: Thank you for the tip.

**Comment by bska** [comment]
jenkins build this opm-material=429 opm-models=678 opm-simulators=3715 please

**Comment by totto82** [comment]
jenkins build this opm-material=429 opm-models=678 opm-simulators=3715 please

**Comment by totto82** [comment]
Thanks for the contribution. I will merge this and the rest.

**Pull request merged**